osu.Game/Rulesets/Objects/Types/IHasRepeats.cs

```diff
@@ -12,5 +15,10 @@ public interface IHasRepeats : IHasEndTime
         /// The amount of times the HitObject repeats.
         /// </summary>
         int RepeatCount { get; }
+
+        /// <summary>
+        /// The samples to be played when each repeat node is hit (0 -> first repeat node, 1 -> second repeat node, etc).
+        /// </summary>
+        List<List<SampleInfo>> RepeatSamples { get; }

```

Comment:
we may want to consider creating a class for this in the future, rather than passing around lists of `SampleInfo`s.